pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,262 @@
+	var policy v1.IOThreadsPolicy
+
+	switch c.ioThreadsPolicy {
+	case string(v1.IOThreadsPolicyAuto):
+		policy = v1.IOThreadsPolicyAuto
+	case string(v1.IOThreadsPolicyShared):
+		policy = v1.IOThreadsPolicyShared
+	default:
+		return params.FlagErr(IOThreadsPolicyFlag, IOThreadErr)
+	}
+
+	instancetypeSpec.IOThreadsPolicy = &policy
+	return nil
+}
+
+func defaultCreateInstancetype() createInstancetype {
+	return createInstancetype{
+		name: "instancetype-" + rand.String(5),
+	}
+}
+
+func (c *createInstancetype) usage() string {
+	return `  # Create a manifest for a Instancetype with a random name and default values:
+  {{ProgramName}} create instancetype
+  
+  # Create a manifest for a Instancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --name my-instancetype --cpu 2
+
+  # Create a manifest for a Instancetype with a specified memory and gpu:
+  {{ProgramName}} create instancetype --memory 256Mi --gpu name:gpu1,devicename:nvidia
+  
+  # Create a manifest for a ClusterInstancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --cluster --name my-instancetype --cpu 2`
+}
+
+func newInstancetype(c *createInstancetype) *instancetypev1alpha2.VirtualMachineInstancetype {
+	return &instancetypev1alpha2.VirtualMachineInstancetype{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "VirtualMachineInstancetype",
+			APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: uint32(c.cpu),
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest: resource.MustParse(c.memory),
+			},
+		},
+	}
+}
+
+func newClusterInstancetype(c *createInstancetype) *instancetypev1alpha2.VirtualMachineClusterInstancetype {
+	return &instancetypev1alpha2.VirtualMachineClusterInstancetype{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "VirtualMachineClusterInstancetype",
+			APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: uint32(c.cpu),
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest: resource.MustParse(c.memory),
+			},
+		},
+	}
+}
+
+func (c *createInstancetype) applyFlags(cmd *cobra.Command, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for flag := range optFns {
+		if cmd.Flags().Changed(flag) {
+			if err := optFns[flag](c, instancetypeSpec); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *createInstancetype) run(cmd *cobra.Command) error {
+	var out []byte
+	var err error
+
+	if c.namespaced {
+		instancetype := newInstancetype(c)
```

Comment:
You might be able to wrap this logic in a generic function if it is not too much effort.